### PR TITLE
Use a volume mount by default to run everything in docker

### PR DESCRIPTION
**Pull request by jameshalsall**
TODO:
- [x] settle on the command API
- [x] update the docs
- [x] test :)
- [x] only set `-T` on `docker-compose exec` commands when needed (pipelines)
- [x] fix the `ws go test integration <test-name>` command
- [x] ensure pipeline integration tests run OK

https://github.com/inviqa/go-sample/pull/16 is a demonstration of this working in pipelines.

This would close #196.

NOTE: I've labelled this as a breaking change, but it shouldn't break any workflows, it should just make them more efficient. The main thing that will change is the `ws recompile` command, which will no longer rebuild the whole app container.

### CODE REVIEW

**Review by jbarton123 — APPROVED** [review]
Lovely work!

File: _twig/docker-compose.yml/application.yml.twig [hunk 1]
```diff
@@ -40,7 +40,5 @@
         @('services.app.environment'),
         @('services.app.environment_secrets'),
      ]), 2, 6) | raw }}
-{% if @('app.mount_volume') %}
     volumes:
       - .:/app
-{% endif %}
```
Comment (andytson-inviqa) [inline _twig/docker-compose.yml/application.yml.twig:46]: if CI tests need data that was in /app as part of the build then this mount will make that inaccessible

Comment (jameshalsall) [reply _twig/docker-compose.yml/application.yml.twig:46]: can you give an example? not sure I understand why that would be the case 🤔

Comment (andytson-inviqa) [reply _twig/docker-compose.yml/application.yml.twig:46]: e.g. `go generate` generating go files in /app that are somehow used in tests

Comment (andytson-inviqa) [reply _twig/docker-compose.yml/application.yml.twig:46]: https://github.com/inviqa/harness-go/blob/406ffa24ae18900320a03b717e9bae829a2cb876/docker/image/app/include/Dockerfile.base.twig#L47 isn't run on a volume, so I'm not sure I get what you mean :)

Comment (andytson-inviqa) [reply _twig/docker-compose.yml/application.yml.twig:46]: I think it'd be safer to not mount the volume in pipeline

Comment (jameshalsall) [reply _twig/docker-compose.yml/application.yml.twig:46]: ah yeah, you are right 🤦🏻 makes more sense not to mount though like you said 👍🏻

**Review by andytson-inviqa — APPROVED** [review]

**Pull request merged**